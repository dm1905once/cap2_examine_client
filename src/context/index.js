import React from 'react';

export const QuestionDetailsContext = React.createContext();
export const AuthContext = React.createContext();