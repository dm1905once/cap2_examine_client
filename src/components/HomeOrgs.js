import React from 'react';
import { Redirect } from 'react-router-dom';
import OrgsLogin from './OrgsLogin';
import OrgsRegister from './OrgsRegister';
import { AuthContext } from "../context";


const HomeOrgs = ( {topMessage=''})=> {
    const { isExaminerAuth, examinerInfo } = React.useContext(AuthContext);

    if ( examinerInfo && isExaminerAuth  ){
        return <Redirect to={`/orgs/${examinerInfo.username}/exams`} />;
    } else {
        return (
            <div className="ui container">
                {topMessage?<div className="ui warning message">{topMessage}</div>:''}
                <div className="ui placeholder segment">
                    <div className="ui two column very relaxed stackable grid">
                        <div className="middle aligned row">
                        <div className="column">
                            <h1>Login</h1>
                            <OrgsLogin />
                        </div>
                        <div className="column">
                            <h1>Register</h1>
                            <OrgsRegister />
                        </div>
                        </div>
                    </div>
                    <div className="ui vertical divider">
                        Or
                    </div>
                </div>
            </div>
        )
    }
}

export default HomeOrgs;