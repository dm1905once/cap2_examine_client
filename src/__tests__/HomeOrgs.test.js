import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import uniqid from 'uniqid';
import HomeOrgs from '../components/HomeOrgs';
import { AuthContext } from "../context";

function renderHomeOrgs(examinerContext){
    return render(
        <AuthContext.Provider value={examinerContext}>
            <HomeOrgs />
        </AuthContext.Provider>
    )
}

// Smoke test
test('HomeOrgs renders without crashing - no examiner logged in', ()=>{
    renderHomeOrgs({isExaminerAuth: false, examinerInfo: undefined});
});

test('HomeOrgs renders without crashing - examiner logged in', ()=>{
    const examinerContext = {
        isExaminerAuth: true, 
        examinerInfo: {
            username: 'test123',
            org_handle: 'org1',
            org_name: 'Some University',
            role: 'examiner'
        }};
    <BrowserRouter>renderHomeOrgs(examinerContext)</BrowserRouter>;
});


// Snapshot test
test('HomeOrgs matches snapshot', ()=>{
    const examinerContext = {isExaminerAuth: false, examinerInfo: undefined};
    const { asFragment } = renderHomeOrgs(examinerContext);
    expect(asFragment()).toMatchSnapshot();
});

// Specialized tests
test('Examiner registration', async ()=>{
    // Select Registration fields
    const examinerHome = renderHomeOrgs({isExaminerAuth: false, examinerInfo: undefined});
    const userNameIn = examinerHome.getAllByPlaceholderText('Username')[1];
    const firstNameIn = examinerHome.getByPlaceholderText('First Name');
    const lastNameIn = examinerHome.getByPlaceholderText('Last Name');
    const emailIn = examinerHome.getByPlaceholderText('Email');
    const pwdIn1 = examinerHome.getByTestId('registerpwd1') 
    const pwdIn2 = examinerHome.getByTestId('registerpwd2') 
    const registerButton = examinerHome.getAllByRole('button')[1];
    expect(userNameIn.toBeInTheDocument);
    expect(pwdIn1.toBeInTheDocument);
    expect(registerButton.toBeInTheDocument);

    // Populate Registration fields
    const userSuffix = uniqid.time();
    await wait(()=>{
        fireEvent.change(userNameIn, { target: { value: `mockname_${userSuffix}` } });
        fireEvent.change(firstNameIn, { target: { value: "TestFirstName" } });
        fireEvent.change(lastNameIn, { target: { value: "TestLastName" } });
        fireEvent.change(emailIn, { target: { value: "TestEmail" } });
        fireEvent.change(pwdIn1, { target: { value: "qwer" } });
        fireEvent.change(pwdIn2, { target: { value: "qwer" } });
    }).then(fireEvent.click(registerButton));
    expect(examinerHome.getByText("Invalid email address")).toBeInTheDocument;

    await wait(()=>{
        fireEvent.change(emailIn, { target: { value: "TestEmail@test.com" } });
    }).then(fireEvent.click(registerButton));
    expect(examinerHome.queryByText("Invalid email address")).not.toBeInTheDocument;

});