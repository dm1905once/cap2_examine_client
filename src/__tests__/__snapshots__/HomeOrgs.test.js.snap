// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeOrgs matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui container"
  >
    <div
      class="ui placeholder segment"
    >
      <div
        class="ui two column very relaxed stackable grid"
      >
        <div
          class="middle aligned row"
        >
          <div
            class="column"
          >
            <h1>
              Login
            </h1>
            <div>
              <form
                class="ui equal width form "
              >
                <div
                  class="fields"
                >
                  <div
                    class="field required"
                  >
                    <label>
                      Username
                    </label>
                    <input
                      name="username"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="fields"
                >
                  <div
                    class="field required"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      name="password"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="ui basic segment"
                >
                  <p>
                     
                  </p>
                  <p>
                     
                  </p>
                  <p>
                     
                  </p>
                </div>
                <button
                  class="ui primary blue right floated right labeled icon button"
                  type="submit"
                >
                  Login
                  <i
                    class="arrow circle right icon"
                  />
                </button>
              </form>
            </div>
          </div>
          <div
            class="column"
          >
            <h1>
              Register
            </h1>
            <div>
              <form
                class="ui equal width form "
              >
                <div
                  class="fields"
                >
                  <div
                    class="field required"
                  >
                    <label>
                      Org Handle
                    </label>
                    <input
                      disabled=""
                      name="org_handle"
                      readonly=""
                      type="text"
                      value="ORG1"
                    />
                  </div>
                  <div
                    class="field required"
                  >
                    <label>
                      Org Key
                    </label>
                    <input
                      disabled=""
                      name="org_key"
                      readonly=""
                      type="text"
                      value="org1"
                    />
                  </div>
                </div>
                <div
                  class="fields"
                >
                  <div
                    class="field required"
                  >
                    <label>
                      Username
                    </label>
                    <input
                      name="username"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="field required"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      data-testid="registerpwd1"
                      name="password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    class="field required"
                  >
                    <label>
                      Re-enter password
                    </label>
                    <input
                      data-testid="registerpwd2"
                      name="password2"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="fields"
                >
                  <div
                    class="field required"
                  >
                    <label>
                      First name
                    </label>
                    <input
                      name="first_name"
                      placeholder="First Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="field"
                  >
                    <label>
                      Last name
                    </label>
                    <input
                      name="last_name"
                      placeholder="Last Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="field required"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      name="email"
                      placeholder="Email"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="ui basic segment"
                >
                  <p>
                     
                  </p>
                  <p>
                     
                  </p>
                  <p>
                     
                  </p>
                </div>
                <button
                  class="ui blue primary right floated right labeled icon button"
                  type="submit"
                >
                  Register
                  <i
                    class="arrow circle right icon"
                  />
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui vertical divider"
      >
        Or
      </div>
    </div>
  </div>
</DocumentFragment>
`;
